lib/routes/movie_list.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+
+import '../data_models/movie.dart';
+import '../repositories/implementation/genres_repository.dart';
+import '../repositories/implementation/movies_repository.dart';
+import '../repositories/interfaces/genres_repository_interface.dart';
+import '../repositories/interfaces/movies_repository_interface.dart';
+import '../routes/movie_details.dart';
+import '../ui_models/movie_preview.dart';
+import '../widgets/row_movie_preview.dart';
+
+class MovieList extends StatefulWidget {
+  const MovieList({super.key});
+
+  static const String routeName = "/";
+
+  @override
+  State<StatefulWidget> createState() => _MovieListState();
+}
+
+class _MovieListState extends State<MovieList> {
+  final MoviesRepositoryInterface moviesRepository = MoviesRepository();
+  final GenresRepositoryInterface genresRepository = GenresRepository();
+  late final Future<List<MoviePreview>> moviePreview;
+
+  @override
+  void initState() {
+    moviePreview = getMovieList();
+    super.initState();
+  }
+
+  Future<List<MoviePreview>> getMovieList() async {
+    List<Movie> movies = await moviesRepository.getMovies();
+    Map<int, String> genres = await genresRepository.getGenres();
+    return movies
+        .map(
+          (movie) => MoviePreview(
+            movie,
+            movie.genres.map((genreId) => genres[genreId]!).toList(),
+          ),
+        )
+        .toList();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        child: FutureBuilder<List<MoviePreview>>(
+          future: moviePreview,
+          builder: (
+            BuildContext context,
+            AsyncSnapshot<List<MoviePreview>> snapshot,
+          ) {
+            if (snapshot.connectionState == ConnectionState.waiting) {
+              return const CircularProgressIndicator();
+            } else if (snapshot.connectionState == ConnectionState.done) {
+              if (snapshot.hasError) {
+                return const Text('Error');
+              } else if (snapshot.hasData) {
+                return ListView.builder(
+                  itemCount: snapshot.data?.length,
+                  itemBuilder: (
+                    BuildContext context,
+                    int index,
+                  ) {
+                    final payload = snapshot.data![index];
+                    return Container(
+                      color: Colors.black54,
+                      width: MediaQuery.of(context).size.width,
+                      child: Card(
+                        color: Colors.black26,
+                        child: GestureDetector(
+                          onTap: () {
+                            Navigator.pushNamed(
+                              context,
+                              MovieDetails.routeName,
+                              arguments: payload,
+                            );
```

Comment:
Nice usage of named routes for navigation! 🚀

Reply:
Thank you! :)